### PR TITLE
Add L2 deploy function

### PR DESCRIPTION
**Pull request by telome**

### CODE REVIEW

File: test/L2Deployment.t.sol [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    UsdsL2Instance inst;
+
+    function setUp() public {
+        inst = UsdsDeploy.deployL2(address(this), l2GovRelay);
+    }
+
+    function testSetUp() public view {
+        assertEq(Usds(inst.usds).wards(l2GovRelay), 1);
+        assertEq(Usds(inst.usds).wards(address(this)), 0);
+        assertEq(Upgrades.getImplementationAddress(inst.usds), inst.usdsImp);
```
Comment (sunbreak1211) [inline test/L2Deployment.t.sol:40]: We have the `getImplementation()` method in the implementation.

Comment (sunbreak1211) [reply test/L2Deployment.t.sol:40]: I see the same is done in the main one. I guess the important thing is keeping consistency. So if updated here and removed the dependency also in `Deployment.t.sol`.

Comment (sunbreak1211) [reply test/L2Deployment.t.sol:40]: Leaving as it is

**Review by sunbreak1211 — APPROVED** [review]
Looks good

**Review by sunbreak1211 — APPROVED** [review]

**Review by oldchili — APPROVED** [review]
LGTM

**Pull request merged**